### PR TITLE
Refactor/cy intercept await

### PR DESCRIPTION
**Pull request by MatthijsSmets**
Custom cypress command has been added to wait for api responses when initializing the app and when switching between tabs, so that tests are more stable and dont crash when a response takes longer than usual.

### CODE REVIEW

File: cypress/e2e/copying.cy.js [hunk 1]
```diff
@@ -1,51 +1,53 @@
...
-    cy.get("[data-cy-nav-tab='testTab']").click();
-    cy.get("#metadataTable tbody", { timeout: 10000 })
-      .find("tr")
-      .should("not.exist");
+  it('Copy report to test tab', () => {
+    cy.navigateToTestTabAndWait();
+    cy.get('#metadataTable tbody', { timeout: 10000 })
+      .find('tr')
+      .should('not.exist');
+    cy.log('Test1234');
```
Comment (philipsens) [inline cypress/e2e/copying.cy.js:20]: I think this is a log used while debugging. Can it be removed now?

Comment (MatthijsSmets) [reply cypress/e2e/copying.cy.js]: Fixed

File: cypress/e2e/testtab_edit.cy.js [hunk 2]
```diff
@@ -1,121 +1,121 @@
...
+    // cy.get('#debug-tree .jqx-tree-dropdown-root > li').should('have.length', 0);
+    cy.navigateToTestTabAndWait();
     // Give UI time to build up the test tab.
-    cy.wait(1000);
-    cy.get("[data-cy-test='selectAll']").click();
-    cy.get("[data-cy-test='deleteSelected']").click();
-    cy.get("[data-cy-delete-modal='confirm']").click();
-    cy.get("#testReports tr", { timeout: 10000 }).should("have.length", 0);
-    cy.get("[data-cy-nav-tab='debugTab']").click();
+    // cy.wait(1000);
```
Comment (philipsens) [inline cypress/e2e/testtab_edit.cy.js:20]: The comments can be removed

File: cypress/e2e/testtab_withOneReport.cy.js [hunk 3]
```diff
@@ -1,115 +1,107 @@
-import chaiColors from "chai-colors";
+import chaiColors from 'chai-colors';
 
 chai.use(chaiColors);
-
-describe("Tests with one report", function () {
-  beforeEach(function () {
+//Hard coded waits have to be replaced with intercepts and waits
```
Comment (philipsens) [inline cypress/e2e/testtab_withOneReport.cy.js:4]: Create an issue for this:

Hard coded waits have to be replaced with intercepts and waits in `cypress/e2e/testtab_withOneReport.cy.js`

Comment (philipsens) [reply cypress/e2e/testtab_withOneReport.cy.js:4]: And remove comment ofc

File: cypress/e2e/copying.cy.js [hunk 4]
```diff
@@ -1,51 +1,52 @@
...
-      .find("tr")
-      .should("not.exist");
-    cy.get("[data-cy-debug='refresh']").click();
-    cy.get("#metadataTable tbody").find("tr").should("have.length", 1);
-    cy.get("[data-cy-debug='selectAll']").click();
-    cy.get("[data-cy-debug='openSelected']").click();
-    cy.get("#debug-tree .jqx-tree-dropdown-root > li").should("have.length", 1);
-    cy.get("[data-cy-debug-editor='copy']").click();
-    cy.get("[data-cy-nav-tab='testTab']").click();
+    //Possibly bad test, it navigates to tab, asserts that the table is empty and then refreshes and asserts that the table is filled.
```
Comment (Matthbo) [inline cypress/e2e/copying.cy.js:21]: Good thing that you note that its a bad test but how does it even work?
Isn't this very time sensitive or even a race condition?

Comment (MatthijsSmets) [reply cypress/e2e/copying.cy.js:21]: Yes you're right, i've added an intercept and await for this

File: cypress/e2e/testtab_withOneReport.cy.js [hunk 5]
```diff
@@ -1,7 +1,3 @@
-import chaiColors from 'chai-colors';
-
-chai.use(chaiColors);
-//Hard coded waits have to be replaced with intercepts and waits
```
Comment (Matthbo) [inline cypress/e2e/testtab_withOneReport.cy.js]: Is this comment still relevant?
It seems a bit out of place

**Review by Matthbo — APPROVED** [review]

File: cypress/e2e/testtab_withOneReport.cy.js (hunk 3 above)
Comment (MatthijsSmets) [reply cypress/e2e/testtab_withOneReport.cy.js]: Comments have been removed

**Review by philipsens — APPROVED** [review]

**Review by philipsens — APPROVED** [review]

**Pull request merged**